android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java

```diff
@@ -285,11 +286,11 @@ private boolean registerClient() throws ExecutionException, InterruptedException
         host = matchingEngine.generateDmeHostAddress();
         if(host == null) {
             Log.e(TAG, "Could not generate host");
-            host = "sdkdemo.dme.mobiledgex.net";   //fallback host
+            host = "wifi.dme.mobiledgex.net";   //fallback host
         }
         port = matchingEngine.getPort(); // Keep same port.
         AppClient.RegisterClientRequest registerClientRequest = matchingEngine.createRegisterClientRequest(ctx,
-                devName, appName, appVersion, carrierName, null, 0, "", "", null);
+                devName, appName, appVersion, carrierName, authToken, cellId, uniqueIdType, uniqueId, tags);
```

Comment:
There's a different Builder style API that doesn't have so many no-ops if the app doesn't actually need them:
createDefaultRegisterClientRequest()